ffi/diplomat/cpp/include/ICU4XUnicodeSetProperty.hpp

```diff
@@ -0,0 +1,85 @@
+#ifndef ICU4XUnicodeSetProperty_HPP
+#define ICU4XUnicodeSetProperty_HPP
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include <algorithm>
+#include <memory>
+#include <variant>
+#include <optional>
+#include <span>
+#include "diplomat_runtime.hpp"
+
+namespace capi {
+#include "ICU4XUnicodeSetProperty.h"
+}
+
+class ICU4XDataProvider;
+struct ICU4XUnicodeSetPropertyResult;
+class ICU4XStaticDataProvider;
+
+/**
+ * A destruction policy for using ICU4XUnicodeSetProperty with std::unique_ptr.
+ */
+struct ICU4XUnicodeSetPropertyDeleter {
+  void operator()(capi::ICU4XUnicodeSetProperty* l) const noexcept {
+    capi::ICU4XUnicodeSetProperty_destroy(l);
+  }
+};
+class ICU4XUnicodeSetProperty {
+ public:
+
+  /**
+   * Gets a set for Unicode property ascii_hex_digit from a [`ICU4XDataProvider`].
+   * See [the Rust docs](https://unicode-org.github.io/icu4x-docs/doc/icu_properties/sets/fn.get_ascii_hex_digit.html) for more information.
+   */
```

Comment:
Is it intentional for the C++ headers to include comments, but not the C headers? It makes sense that the copies of the C headers in `ffi/diplomat/cpp` are just an implementation detail of the C++ headers, but the same argument doesn't seem to hold for `ffi/diplomat/c`.

Reply:
@gregtatum added C++ docs generation in https://github.com/rust-diplomat/diplomat/pull/85.  The primary documentation is intended to be the `*.rst` files, but perhaps we should put the docs into the `*.h` files as well. https://github.com/rust-diplomat/diplomat/issues/101